driverprocessresults/main.py

```diff
@@ -0,0 +1,208 @@
+        split = os.path.dirname(session_path)
+        run_name = os.path.basename(split)
+        all_stats_dir = cls._create_stats_dir(session_path)
+        session_stats_dir = all_stats_dir / run_name / session_name
+        
+        if delete_previous and session_stats_dir.is_dir():
+            shutil.rmtree(session_stats_dir)
+        
+        if not session_stats_dir.is_dir():
+            os.makedirs(session_stats_dir)
+            #HERE
+            function_times = {}
+            files = os.listdir(Path(session_path))
+            for file in files: 
+                if '.csv' in file:
+                    #creating fp=sess/file.csv
+                    fp = os.path.join(session_path, file)
+                    #opening csv file
+                    with open(fp) as f:
+                        #scan over all lines in file
+                        for i, line in enumerate(f):
+                            #QUESTION: ask for an explanation
+                            #does this created 2 keys and assigns all values to the 2 keys?
+                            vals = line.split(',')
+                            if vals[1] in function_times.keys():
+                                function_times[vals[1]].append(float(vals[2]))
+                            else:
+                                function_times[vals[1]] = [float(vals[2])]
+                else:
+                    if verbose: #QUESTION: what is the meaning? right now verbose will always be false?
+                        print(file) #why do we print file()?
+            if verbose:
+                print('Min {0}'.format(min(function_times['client()'])))
+                print('Max {0}'.format(max(function_times['client()'])))
+            try:
+                if "run_model" in function_times:
+                    #will always be false and therefore skip this?
+                    if verbose:
+                        #Question: explanation?
+                        num_run = len(function_times['run_model'])
+                        print(f'there are {num_run} values in the run_model entries')
+                    cls._make_hist_plot(function_times['run_script'], 'run_script()', 'run_script.pdf', session_stats_dir)
+                    cls._make_hist_plot(function_times['run_model'], 'run_model()', 'run_model.pdf', session_stats_dir)
+                if "client()" in function_times:
+                    cls._make_hist_plot(function_times['client()'], 'client()', 'client_constructor_dist.pdf', session_stats_dir)
+                
+                if "put_tensor" in function_times:
+                    cls._make_hist_plot(function_times['put_tensor'], 'put_tensor()', 'put_tensor.pdf', session_stats_dir)
+
+                if "unpack_tensor" in function_times:
+                    cls._make_hist_plot(function_times['unpack_tensor'], 'unpack_tensor()', 'unpack_tensor.pdf', session_stats_dir)
+
+                if "get_list" in function_times:
```

Comment:
Seems like there should be a better way to do these than hard-coding each potential name. Failing that, making a list of the names and looping through it would eliminate a lot of duplicate code here

Reply:
done!